ext/proxy/Dxxxx.md

```diff
@@ -1135,63 +1089,91 @@ Change 24.6 "Header `<experimental/ranges/iterator>` synopsis" ([iterator.synops
 > ```
 
-Before subsubsection "Iterator associated types" ([iterator.assoc]), add a
-new subsubsection "Iterator utilities" ([iterator.utils]). Under that
-subsubsection, insert the following:
+Before subsubsection "Iterator associated types" ([iterator.assoc]), add a new subsubsection "Iterator utilities" ([iterator.utils]). Under that subsubsection, insert the following:
 
-> > ```c++
-> > template <class I>
-> >   requires _Dereferenceable<I>
-> > auto iter_move(I&& r) noexcept(see below) -> see below;
-> > ```
+> 24.6.? *iter_move*                     [iterator.utils.iter_move]
 >
-> 1\. The return type is `Ret` where `Ret` is
-> `remove_reference_t<reference_t<I>>&&` if `I` is
-> a reference type; `decay_t<I>`, otherwise.
-> 2\. The expression in the `noexcept` is equivalent to:
-> > ```c++
-> > noexcept(Ret(std::move(*r)))
-> > ```
-> 
-> 3\. *Returns:* `std::move(*r)`
+> 1\. The name `iter_move` denotes a *customization point object* (17.5.2.1.5 [customization.point.object]). The effect of the expression `ranges::iter_move(E)` for some expression `E` is equivalent to `iter_move(E)`, with overload resolution performed in an unspecified context described below.
 >
-> <span style="color:blue">[*Editorial note:* -- Future work: Rather than defining a new `iter_swap` in namespace `std::experimental::ranges::v1`, it will probably be necessary to constrain the `iter_swap` in namespace `std` much the way the Ranges TS constrains `std::swap`. --*end note*]</span>
+> 2\. The context in which the expression `iter_move(E)` is evaluted
 >
-> > ```c++
-> > template <class I1, class I2,
-> >   Readable _R1 = remove_reference_t<I1>,
-> >   Readable _R2 = remove_reference_t<I2>>
-> >   requires Swappable<reference_t<_R1>, reference_t<_R2>>()
-> > void iter_swap(I1&& r1, I2&& r2)
-> >   noexcept(is_nothrow_swappable_with<reference_t<_R1>, reference_t<_R2>>::value);
-> > ```
+> > -- does not include a declaration of `ranges::iter_move`,
+> >
+> > -- includes the lookup set produced by argument-dependent lookup ([basic.lookup.argdep]), and
+> >
+> > -- includes an `iter_move` function template equivalent to the following:
+> >
+> > > ```c++
+> > > // Exposition only
+> > > template <_Dereferenceable I>
+> > > *see below* iter_move(I&& r) noexcept(*see below*);
+> > > ```
+> > > > 2.1. The return type is `Ret` where `Ret` is
+> > > > `remove_reference_t<reference_t<I>>&&` if 
+> > > > `reference_t<I>` is a reference type;
+> > > > `decay_t<reference_t<I>>`, otherwise.
+> > > >
+> > > > 2.2. The expression in the `noexcept` is equivalent to:
+> > > > > ```c++
+> > > > > noexcept(Ret(std::move(*r)))
+> > > > > ```
+> > > >
+> > > > 2.3. *Effects:* Equivalent to `std::move(*r)`
 >
-> 4\. *Effects*: `swap(*r1, *r2)`
+> 3\. If the `iter_move` function selected by overload resolution ([over.match]) does not return a (possibly proxy) reference to, or copy of, the object denoted by `*r`, the program is ill-formed with no diagnostic required.
```

Comment:
We have `{ iter_move(i) } -> Same<rvalue_reference_t<I>>;` in `Readable`; that's enough to establish equality-preservation.